store/sql_team_store.go

```diff
@@ -583,6 +583,32 @@ func (s SqlTeamStore) GetTeamsForUser(userId string) StoreChannel {
 	return storeChannel
 }
 
+func (s SqlTeamStore) GetTeamsUnreadForUser(teamId, userId string) StoreChannel {
+	storeChannel := make(StoreChannel, 1)
+
+	go func() {
+		result := StoreResult{}
+
+		var members []*model.TeamUnread
+		_, err := s.GetReplica().Select(&members,
```

Comment:
Could we use a query that gets all of their ChannelMembers and then sums them up for each team in the Go code? That would remove a lot of strain from the DB and shouldn't be very expensive on the Go side